Sources/FileManager+Testable.swift

```diff
@@ -11,23 +11,41 @@
 //
 
 @testable import AEPServices
-import Foundation
 
 public extension FileManager {
-    func clearCache() {
-        let knownCacheItems: [String] = ["com.adobe.edge", "com.adobe.edge.identity", "com.adobe.edge.consent"]
+    /// Clears the cache for specified items within the application's cache directory. Ex: the event database for a given extension.
+    ///
+    /// This method removes cache items based on the provided list of cache item names and directory flags. If no list is provided,
+    /// a default set of cache items is used. This operation is intended to clear cached data related to specific extensions or
+    /// functionalities within an app, such as consent, identity, and disk caches. The method handles both file and directory types
+    /// of cache items.
+    ///
+    /// - Parameter cacheItems: An optional array of tuples where each tuple contains the `name` of the cache item (as a `String`)
+    ///   and a `Bool` indicating whether the cache item is a directory (`true`) or a file (`false`). If `nil`, a default list of
+    ///   cache items is used.
+    func clearCache(_ cacheItems: [(name: String, isDirectory: Bool)]? = nil) {
+        let LOG_TAG = "FileManager"
+
+        // Use caller provided values, defaults otherwise.
+        let cacheItems = cacheItems ?? [(name: "com.adobe.edge", isDirectory: false),
+                                        (name: "com.adobe.edge.consent", isDirectory: false),
+                                        (name: "com.adobe.edge.identity", isDirectory: false),
+                                        (name: "com.adobe.eventHistory", isDirectory: false),
+                                        (name: "com.adobe.mobile.diskcache", isDirectory: true),
+                                        (name: "com.adobe.module.signal", isDirectory: false),]
         guard let url = self.urls(for: .cachesDirectory, in: .userDomainMask).first else {
+            Log.warning(label: LOG_TAG, "Unable to find valid cache directory path.")
             return
         }
 
-        for cacheItem in knownCacheItems {
+        for cacheItem in cacheItems {
             do {
-                try self.removeItem(at: URL(fileURLWithPath: "\(url.relativePath)/\(cacheItem)"))
+                try self.removeItem(at: URL(fileURLWithPath: "Library/Caches/\(cacheItem.name)", isDirectory: cacheItem.isDirectory))
```

Comment:
Per your comment on the logged errors, is it possible to check if the directory exists before attempting to delete it so an error isn't logged?

Reply:
Apple recommends against checking for file existing when predicating behavior: https://developer.apple.com/documentation/foundation/filemanager/1410277-fileexists

> Attempting to predicate behavior based on the current state of the file system or a particular file on the file system is not recommended. Doing so can cause odd behavior or race conditions. It’s far better to attempt an operation (such as loading a file or creating a directory), check for errors, and handle those errors gracefully than it is to try to figure out ahead of time whether the operation will succeed. For more information on file-system race conditions, see [Race Conditions and Secure File Operations](https://developer.apple.com/library/archive/documentation/Security/Conceptual/SecureCodingGuide/Articles/RaceConditions.html#//apple_ref/doc/uid/TP40002585) in [Secure Coding Guide](https://developer.apple.com/library/archive/documentation/Security/Conceptual/SecureCodingGuide/Introduction.html#//apple_ref/doc/uid/TP40002415).